homeassistant/components/advantage_air/__init__.py

```diff
@@ -50,19 +51,25 @@ async def async_get():
         update_interval=timedelta(seconds=ADVANTAGE_AIR_SYNC_INTERVAL),
     )
 
-    async def async_change(change):
-        try:
-            if await api.async_change(change):
-                await coordinator.async_refresh()
-        except ApiError as err:
-            _LOGGER.warning(err)
+    def error_handle_factory(func):
+        """Return the provided API function wrapped in an error handler and coordinator refresh."""
+
+        async def error_handle(param):
+            try:
+                if await func(param):
+                    await coordinator.async_refresh()
+            except ApiError as err:
+                _LOGGER.warning(err)
```

Comment:
Usually we re-raise as `HomeAssistantError`

https://developers.home-assistant.io/docs/integration_quality_scale_index/#silver-
Operations like service calls and entity methods (e.g. Set HVAC Mode) have proper exception handling. Raise ValueError on invalid user input and raise HomeAssistantError for other failures such as a problem communicating with a device.

Reply:
Do you want me to address that here or as a seperate PR. This code block is functionally the same as whats currently in core and this change isnt related to the lights platform addition.

Reply:
If you want to change it now or do it in a followup, either WFM

Reply:
I'd prefer to not complicate this PR further given its age, so Ill raise a code quality PR to address this.